src/client/Client.cc

```diff
@@ -8601,12 +8625,14 @@ int Client::_sync_fs()
 {
   ldout(cct, 10) << "_sync_fs" << dendl;
 
-  // wait for unsafe mds requests
-  // FIXME
-  
   // flush caps
   flush_caps();
-  wait_sync_caps(last_flush_tid);
+  ceph_tid_t flush_tid = last_flush_tid;
+
+  // wait for unsafe mds requests
+  wait_unsafe_requests();
+
+  wait_sync_caps(flush_tid);
 
   // flush file data
   // FIXME
```

Comment:
On Wed, Oct 28, 2015 at 8:10 AM, Gregory Farnum <notifications@github.com>
wrote:

> In src/client/Client.cc
> <https://github.com/ceph/ceph/pull/6388#discussion_r43205480>:
>
> >    // flush caps
> >    flush_caps();
> > -  wait_sync_caps(last_flush_tid);
> > +  ceph_tid_t flush_tid = last_flush_tid;
> > +
> > +  // wait for unsafe mds requests
> > +  wait_unsafe_requests();
> > +
> > +  wait_sync_caps(flush_tid);
> >
> >    // flush file data
> >    // FIXME
>
> So sync_fs actually still doesn't work, since we aren't waiting on OSD
> requests? Should fix that up while doing the rest of it.
>

I added code that flush objectcacher. but sync_fs() still does not wait for
unsafe sync write.



> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ceph/ceph/pull/6388/files#r43205480>.
>